packages/protobuf/src/private/util-common.ts

```diff
@@ -55,18 +54,37 @@ export function makeUtilCommon(): Omit<Util, "newFieldList" | "initFields"> {
               !(val instanceof sourceField.T)
             ) {
               val = new sourceField.T(val);
+            } else if (
+              sourceField &&
+              sourceField.kind === "scalar" &&
+              sourceField.T === ScalarType.BYTES
+            ) {
+              val = toU8Arr(val);
             }
             t[localName] = { case: sk, value: val };
             break;
           case "scalar":
           case "enum":
-            t[localName] = s[localName];
+            let target = s[localName];
+            if (member.T === ScalarType.BYTES) {
+              target = member.repeated
+                ? (target as ArrayLike<number>[]).map(toU8Arr)
+                : toU8Arr(target);
+            }
+            t[localName] = target;
             break;
           case "map":
             switch (member.V.kind) {
               case "scalar":
               case "enum":
-                Object.assign(t[localName], s[localName]);
+                let target = s[localName];
+                if (member.V.T === ScalarType.BYTES) {
+                  target = {};
+                  for (const [k, v] of Object.entries(s[localName])) {
+                    target[k] = toU8Arr(v as ArrayLike<number>);
+                  }
+                }
+                Object.assign(t[localName], target);
```

Comment:
We should avoid modifying the input. What about

```suggestion
                if (member.V.T === ScalarType.BYTES) {
                  for (const [k, v] of Object.entries(s[localName])) {
                    t[localName][k] = toU8Arr(v as ArrayLike<number>);
                  }
                } else {
                  Object.assign(t[localName], s[localName]);
                }
```

?

Reply:
We are not modifying the response, we reset target at line 82 to `{}`

Reply:
I missed that one too 🙈. Discussed offline, we went with the one with smaller bundle size.